packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -184,24 +187,32 @@ export const useSignInContext = (): SignInContextType => {
   };
 };
 
+type PagesType = {
+  routes: NavbarRoute[];
+  contents: CustomPageContent[];
+  isAccountPageRoot: boolean;
+  pageToRootNavbarRouteMap: Record<string, NavbarRoute>;
+};
+
 export type UserProfileContextType = UserProfileCtx & {
   queryParams: ParsedQs;
   authQueryString: string | null;
+  pages: PagesType;
 };
 
-// UserProfile does not accept any props except for
-// `routing` and `path`
-// TODO: remove if not needed during the components v2 overhaul
 export const useUserProfileContext = (): UserProfileContextType => {
-  const { componentName, ...ctx } = (React.useContext(ComponentContext) || {}) as UserProfileCtx;
+  const { componentName, customPages, ...ctx } = (React.useContext(ComponentContext) || {}) as UserProfileCtx;
   const { queryParams } = useRouter();
 
   if (componentName !== 'UserProfile') {
     throw new Error('Clerk: useUserProfileContext called outside of the mounted UserProfile component.');
   }
 
+  const pages = useMemo(() => createCustomPages(customPages || []), [customPages]);
```

Comment:
You are right! We cannot hook on the customPages changes, because we can't really be 100% sure when custom pages contents change. So, I may just remove the `useMemo` here.